test/Client.test.js

```diff
@@ -0,0 +1,335 @@
+import Client from '../src/Client';
+import SocketHandler from '../src/core/SocketHandler';
+import Api from '../src/core/Api';
+import { Event } from '../src/core/ChainEvent';
+import debug from 'debug';
+import { find, isEqual } from 'lodash';
+
+const log = debug('log-test');
+const options = {
+  accounts: 3,
+  block_mine_time: 0,
+  clean_on_stop: false
+};
+
+let client;
+
+const _create = async (options) => {
+  client.create(options);
+  const {
+      payload: {
+        response: { id }
+      }
+    } = await client.once('api', Event.CHAIN_CREATED);
+
+    if (options.step_id) {
+      return client.sequenceEvents(id, [
+        Event.CHAIN_STARTED,
+        Event.CHAIN_DEPLOYING,
+        Event.CHAIN_STATUS_ACTIVE,
+        Event.CHAIN_DEPLOYED,
+        Event.CHAIN_READY
+      ]);
+    } else {
+      return client.sequenceEvents(id, [
+        Event.CHAIN_STARTED,
+        Event.CHAIN_STATUS_ACTIVE,
+        Event.CHAIN_READY
+      ]);
+    }
+};
+
+const _stop = ( id ) => {
+  client.stop(id);
+  return client.sequenceEvents(id, [
+    Event.OK,
+    Event.CHAIN_STATUS_TERMINATING,
+    Event.CHAIN_TERMINATED
+  ]);
+};
+
+const _restart = ( id ) => {
+  client.restart(id);
+  return client.sequenceEvents(id, [
+    Event.CHAIN_STARTED,
+    Event.CHAIN_READY,
+    Event.CHAIN_STATUS_ACTIVE
+  ]);
+};
+
+const _takeSnapshot = (id, description) => {
+  client.takeSnapshot(id, description);
+  return client.sequenceEvents(id, [
+    Event.CHAIN_STATUS_TAKING_SNAP,
+    Event.SNAPSHOT_TAKEN,
+    Event.CHAIN_STATUS_SNAP_TAKEN,
+    Event.CHAIN_STATUS_ACTIVE
+  ]);
+};
+
+const _restoreSnapshot = async (id, snapshot) => {
+  const { data: list } = await client.api().listAllChains();
+  const exists = find(list, { id });
+
+  client.restoreSnapshot(id, snapshot);
+
+  if (!exists) {
+    return client.sequenceEvents('api', [
+      Event.CHAIN_CREATED,
+      Event.CHAIN_STARTED
+    ]);
+  } else {
+    return client.sequenceEvents(id, [
+      Event.OK,
+      Event.CHAIN_STATUS_REVERTING_SNAP,
+      Event.SNAPSHOT_REVERTED,
+      Event.CHAIN_STATUS_SNAP_REVERTED,
+      Event.CHAIN_STATUS_ACTIVE
+    ]);
+  }
+};
+
+beforeEach(() => {
+  client = new Client();
+});
+
+afterEach(async () => {
+  const { data: list } = await client.api().listAllChains();
+  for (const chain of list) {
+    const { id } = chain;
+    await client.delete(id);
+  }
+});
+
+test('client will be created correctly', () => {
+  expect(client.socket() instanceof SocketHandler).toBeTruthy();
+  expect(client.api() instanceof Api).toBeTruthy();
+});
+
+test('client will initialise socket connection', async () => {
+  expect(client.socket().connected()).toBeFalsy();
+  await client.init();
+  expect(client.socket().connected()).toBeTruthy();
```

Comment:
similar thing here, if the return value of `client.socket().connected()` somehow changed to a response like `{msg: not connected}` this test would still pass. I would assert `.toBe(true)` because it makes clear exactly what we're expecting.